### DISCUSSION TITLE
external program

### DISCUSSION THREAD

**Discussion by adj00080** [artisan-roaster-scope/artisan#824] [Q&A]
I'm having trouble using external program as the device. I followed the example in the help window and I am getting the following when I press the ON button " callprogram() received @line 753". When I am in idle, I am getting the following "Websocket connection failed [WinEror 10061] No connection could be made because the target machine actively refused it". 

Is there anything else I need to enable from a pc side or other settings besides the target open path I need to update?

**Comment by bigl00z3** [comment]
You should not need anything else. 
Try the function with something simple first. If you can see that Artisan is communicating with your PC, then the issue is more with the command to your program.

The best way to test is probably to try with a talking alarm:
[https://artisan-roasterscope.blogspot.com/2017/12/talking-alarms.html](https://artisan-roasterscope.blogspot.com/2017/12/talking-alarms.html)

You can start by typing the speak command in your computer terminal and if it works, then try from Artisan. Once this work, try your program.

Nico

**Reply by adj00080** [reply]
Thanks Nico, I will try this out. 

I was able to get my system up and running using websockets thanks to the code from glassTransition on forum#701[https://github.com/artisan-roaster-scope/artisan/discussions/701]. 

I think websockets might work better for my application. I am using an older raspberry pi with an adafruit thermocouple amplifier. The pi was a bit slow when trying to run the software but can handle being a simple web server without trouble. 

Here's the code i ended up with: *(I added a thermocouple reading and the IP so i can access wirelessly, the rest is from glassTransition's code.)
[artisan_websocket.txt](https://github.com/artisan-roaster-scope/artisan/files/8174883/artisan_websocket.txt)
